run.py

```diff
@@ -409,31 +411,80 @@ def run_diffusion_processsing(**args):
-        # print(acqs)
-        # posData = []
-        # negData = []
-        # for acq in acqs:
-        #     pos = "EMPTY"
-        #     neg = "EMPTY"
-        #     dwis = layout.get(subject=subject_label,
-        #                       type='dwi', acquisition=acq,
-        #                       extensions=["nii.gz", "nii"])
-        #     assert len(dwis) <= 2
-        #     for dwi in dwis:
-        #         dwi = dwi.filename
-        #         if "-" in layout.get_metadata(dwi)["PhaseEncodingDirection"]:
-        #             neg = dwi
-        #         else:
-        #             pos = dwi
-        #     posData.append(pos)
-        #     negData.append(neg)
-        #
-        # print(negData)
-        # print(posData)
+        posData = []
+        negData = []
+        PEdir = "None"
+        dwiname = "Diffusion"
+        dirnums = []
+        onerun = False
+
+        numruns = set(layout.get(target='run', return_type='id',
+                                 subject=subject_label, type='dwi',
+                                 extensions=["nii.gz", "nii"]))
+        # accounts for multiple runs, number of directions, and phase encoding directions
+        if not numruns:
+            onerun= True
+            numruns = {'run-01'}
+        if numruns:
+            for session in numruns:
+                if not onerun:
+                    bvals = [f.filename for f in layout.get(subject=subject_label,
+                                                            type='dwi', run=session,
+                                                            extensions=["bval"])]
+                else:
+                    bvals = [f.filename for f in layout.get(subject=subject_label,
+                                                            type='dwi', extensions=["bval"])]
+                dwi_dict = {'bvalFile':[], 'bval':[], 'dwiFile':[], 'direction':[]}
+                for bvalfile in bvals:
+                    with open(bvalfile) as f: # get number of directions
+                        bvalues = [bvalue for line in f for bvalue in line.split()]
+                    dwi_dict['bvalFile'].append(bvalfile)
+                    dwi_dict['bval'].append(len(bvalues) - 1)
+                    dwiFile = glob(os.path.join(os.path.dirname(bvalfile),'{0}.nii*'.format(os.path.basename(bvalfile).split('.')[0]))) # ensures bval file has same name as dwi file
+                    assert len(dwiFile) == 1
+                    dwi_dict['dwiFile'].append(dwiFile[0])
+                    dwi_dict['direction'].append(layout.get_metadata(dwiFile[0])["PhaseEncodingDirection"][0])
+
+                # check if length of lists in dictionary are the same
+                n = len(dwi_dict['bvalFile'])
+                assert all(len(dwi_dict[k]) for k,v in dwi_dict.items())
+
+                for dirnum in set(dwi_dict['bval']):
+                    idxs = { i for k,v in dwi_dict.iteritems() for i in range(0,len(dwi_dict['bval'])) if v[i] == dirnum }
```

Comment:
Line 450: I made a mistake - it's supposed to be 
assert all(len(dwi_dict[k]) ==n for k,v in dwi_dict.items())

Line 453: extracts the index values in dwi_dict['bval'] if the value matches "dirnum", which is the number of directions (i.e. 98 or 99). These index values gets used to find the PE directions, dwi files names, etc. in the dictionary.